### ISSUE TITLE
Blend issues with impeller 

### ISSUE THREAD

**Issue by esDotDev** [gskinnerTeam/flutter-wonderous-app#34]
Using the latest master, with code like:
```
Image(
  image: ...,
  color: Colors.black,
  colorBlendMode: BlendMode.color,
  opacity: AlwaysStoppedAnimation(0.2),
),
```
We are seeing blending issues: 
![image](https://user-images.githubusercontent.com/736973/191562555-d41f12e4-16d0-4a26-9953-6b4d07d28707.png)

Without impeller: 
![image](https://user-images.githubusercontent.com/736973/191562395-8de13e5c-73ca-49ef-a242-9a1e89c3d65b.png)

**Comment by zanderso** [comment]
cc @bdero

**Comment by bdero** [comment]
This looks like the alpha channel is either getting dropped or ignored somewhere along the line for these images. Trying to reproduce this.

**Comment by bdero** [comment]
I believe the issue is the way I did the alpha compositing for advanced blends in the case where a foreground color is specified. In this case, the source color is being overridden to opaque black, so the fragment alpha output is always 1: https://github.com/flutter/engine/blob/main/impeller/entity/shaders/blending/advanced_blend.glsl#L43

I'm currently working out the correct behavior for this.

**Comment by bdero** [comment]
Fix: https://github.com/flutter/engine/pull/36338.

BTW, that overhang shadow isn't missing in Impeller, right? I assume it fades in when starting to scroll down.

**Comment by esDotDev** [comment]
Correct, the shadow is good w/ impeller, its just not visible in that screenshot.

**Comment by esDotDev** [comment]
Fixed in `2e0d6f100c9ea77d2b31b04752fa4e185b1a05cf`
